cwms_radar_api/src/main/java/cwms/radar/data/dao/BasinDao.java

```diff
@@ -0,0 +1,98 @@
+package cwms.radar.data.dao;
+
+import cwms.radar.api.enums.Unit;
+import cwms.radar.api.enums.UnitSystem;
+import cwms.radar.data.dto.basinconnectivity.Basin;
+import cwms.radar.data.dto.basinconnectivity.Stream;
+import org.jooq.DSLContext;
+import usace.cwms.db.jooq.dao.CwmsDbBasinJooq;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BasinDao extends JooqDao<Basin>
+{
+    public BasinDao(DSLContext dsl)
+    {
+        super(dsl);
+    }
+
+    public List<Basin> getAllBasins(String unitSystem, String officeId) throws SQLException
```

Comment:
We're going to want to add pagination to this. I can imagine this list getting pretty long at some point. 
But I'm okay with dealing with it a little later down the line.